packages/schema/src/types/components.ts

```diff
@@ -16,6 +16,11 @@ export interface Component {
    * components will be used instead of lower priority components.
    */
   priority?: number
+  /**
+   * This prevent the component to be transformed with a `createClientOnly()` client-side if the mode is `client`
+   * An example is the `<NuxtClientFallback>` component which handles it's hydration client-side
+   */
+  noClientOnlyTransform?: boolean
```

Comment:
I'm not sure about that 🤔 I guess we'd need to know if some users need it. It's mainly a way to keep side-effects modules only on 1 side (server or client).

Reply:
I'd be happy to merge an fix needing this with an internal property that we set and consume for `<NuxtClientFallback>`, rather than exposing this as a public property for components which can be enabled by users ...

Reply:
let's mark it private for now :+1: